kitsune/wiki/facets.py

```diff
@@ -69,31 +73,36 @@ def documents_for(locale, topics=None, products=None):
     return documents, fallback_documents
 
 
-def _documents_for(locale, topics=None, products=None):
-    """Returns a list of articles that apply to passed in topics and products."""
+def _documents_for(locale, topics=None, products=None, current_document=None):
+    """Returns a list of articles that apply to passed in locale, topics and products."""
     # First try to get the results from the cache
     cache_key = _cache_key(locale, topics, products)
     documents_cache_key = f"documents_for:{cache_key}"
     documents = cache.get(documents_cache_key)
+    exclude_current_document = isinstance(current_document, Document)
     if documents is not None:
+        if exclude_current_document:
```

Comment:
I moved the exclusion here b/c of [this call](https://github.com/mozilla/kitsune/pull/5483/files#diff-754fad9f530ba68d2f0722f3572f6f0b45752ff296970f1839e01c660fbfd80fR63) to `_documents_for` . This way we are making sure that we are always excluding the document.